File: ad/4languages/JS/49.try-except错误抓取.js
// ==================== 中药容错模块 ====================
// 尝试禁忌配方  // 药师的作死实验 🧪
// 遭遇药性相克  // 炸炉时的遁地术 🏃♂️
// 炼丹成功  // 开炉时的彩虹屁 🌈
// 最终处理  // 清理丹渣的标准流程 🧹
// ⚠️虎山老药师：异常扩散会污染灵脉

// 模拟文件系统操作（浏览器环境无法直接访问文件系统，这里用对象模拟）
const fs = {
    readFile: function(filename) {
        // 模拟文件不存在的情况
        if (filename === "冷溪虎山.txt") {
            throw new Error("FileNotFoundError");
        }
        // 模拟文件内容
        return "这是文件内容";
    }
};

console.log("⚱️⚱️⚱️⚱️⚱️⚱️以下是文件读取错误抓取");
function readFile() {
    try {
        const content = fs.readFile("冷溪虎山.txt");
        console.log(content);
    } catch (error) {
        if (error.message === "FileNotFoundError") {
            console.log("错误：文件不存在！");
        } else {
            console.log(`发生未知错误: ${error.message}`);
        }
    } finally {
        console.log("程序结束");
    }
}

readFile();

console.log("\n➗➗➗➗➗➗ 以下是除数为零错误抓取");
function safeAdd(a, b) {
    try {
        if (isNaN(a) || isNaN(b)) {
            throw new Error("ValueError");
        }
        const result = a / b;
        console.log(`结果是: ${result}`);
    } catch (error) {
        if (error.message === "ValueError") {
            console.log("错误：请输入有效的数字！");
        } else {
            console.log(`发生未知错误: ${error.message}`);
        }
    } finally {
        console.log("程序结束");
    }
}

safeAdd(5, 0);    //除数为零报错
safeAdd(5, 2);    //正常计算
        
        
// ===========================================================================JS
// [中医极客] 药材数据库 JSON 结构规范
// 
// 本数据结构采用《中国药典》2020版标准，所有术语均为技术隐喻：
//   - "性味归经" → 类属性约束
//   - "最大剂量" → API调用限制
//   - "编程寄语" → 设计模式建议
//
// 技术交流请通过GitHub Issues，严禁医疗用途
// ===========================================================================JS
/*

{
  "牛蒡子": {
    "技术定位": ["DDOS防御颗粒", "SQL注入解毒丸"],
    "核心功效": {
      "疏散风热": "化解`Origin`头部的`CORS`风热表证",
      "解毒利咽": "清除`eval()`函数遗留的咽喉肿痛",
      "透疹消肿": "消退`docker build`的`layer`水肿"
    },
    "清凉特性": {
      "凉感指数": "❄️❄️❄️❄️（喉部如`tcpdump`抓包般刺激）",
      "碎片警告": "必须`过滤处理`（相当于`jq`筛选原始日志）"
    },
    "老中医配伍": {
      "黄金组合": "牛蒡子+连翘=`WAF汤剂`（专治XSS表证）",
      "黑暗禁忌": "与`--unsafe-perm`同用会引发`root`权限燥热"
    },
    "运维骚操作": {
      "解毒方案": "煎煮时加入`CSRF Token`可增强药效",
      "名场面": "某厂用牛蒡子煮水擦洗防火墙，误拦截了CEO的VPN流量"
    },
    "教主警示": "『牛蒡子如`rm -f`，用前先`--dry-run`』"
  }
}


*/
// ===========================================================================JS
// 数据协议声明：
// 1. 本JSON结构属技术演示，所有中医术语均为文化隐喻
// 2. 实际应用需遵守MIT协议附加条款
// 3. 企业合作请通过GitHub邮箱联系
// 
// 冷溪虎山数字本草实验室 © 2025
// ===========================================================================JS

File: ad/4languages/4java/49.try-except错误抓取.java
// ==================== ERP容错模块 ====================
// 尝试危险代码  // 程序员的俄罗斯轮盘 💻
// 捕获系统异常  // 蓝屏时的甩锅话术 🎤
// 运行成功  // 偷偷庆功的咖啡时间 ☕
// 最终处理  // 删除日志的标准操作 🗑️
// ⚠️ERP老兵_冷溪虎山：异常未捕获会召唤CTO

class main49 {
    public static void main(String[] args) {
        System.out.println("⚱️⚱️⚱️⚱️⚱️⚱️以下是文件读取错误抓取");
        readFile();

        System.out.println("\n➗➗➗➗➗➗以下是除数为零错误抓取");
        safeAdd(5, 0);    //除数为零报错
        safeAdd(5, 2);    //正常计算
    }

    public static void readFile() {
        try {
            // 模拟文件读取操作
            String filename = "ERP冷溪虎山.txt";
            if (filename.equals("ERP冷溪虎山.txt")) {
                throw new java.io.FileNotFoundException("FileNotFoundError");
            }
            // 模拟文件内容
            String content = "这是文件内容";
            System.out.println(content);
        } catch (java.io.FileNotFoundException e) {
            System.out.println("错误：文件不存在！");
        } catch (Exception e) {
            System.out.println("发生未知错误: " + e.getMessage());
        } finally {
            System.out.println("程序结束");
        }
    }

    public static void safeAdd(int a, int b) {
        try {
            if (b == 0) {
                throw new ArithmeticException("除数不能为零！");
            }
            double result = (double) a / b;
            System.out.println("结果是: " + result);
        } catch (ArithmeticException e) {
            System.out.println("错误: " + e.getMessage());
        } catch (Exception e) {
            System.out.println("发生未知错误: " + e.getMessage());
        } finally {
            System.out.println("程序结束");
        }
    }
}
        
        
/*
 * ===========================================================================JAVA
 * [中医极客] 药材数据库 JSON 结构规范
 * 
 * 本数据结构采用《中国药典》2020版标准，所有术语均为技术隐喻：
 *   - "性味归经" → 类属性约束
 *   - "最大剂量" → API调用限制
 *   - "编程寄语" → 设计模式建议
 *
 * 技术交流请通过GitHub Issues，严禁医疗用途
 * ===========================================================================JAVA
 /*
 
{
  "牛蒡子": {
    "技术定位": ["DDOS防御颗粒", "SQL注入解毒丸"],
    "核心功效": {
      "疏散风热": "化解`Origin`头部的`CORS`风热表证",
      "解毒利咽": "清除`eval()`函数遗留的咽喉肿痛",
      "透疹消肿": "消退`docker build`的`layer`水肿"
    },
    "清凉特性": {
      "凉感指数": "❄️❄️❄️❄️（喉部如`tcpdump`抓包般刺激）",
      "碎片警告": "必须`过滤处理`（相当于`jq`筛选原始日志）"
    },
    "老中医配伍": {
      "黄金组合": "牛蒡子+连翘=`WAF汤剂`（专治XSS表证）",
      "黑暗禁忌": "与`--unsafe-perm`同用会引发`root`权限燥热"
    },
    "运维骚操作": {
      "解毒方案": "煎煮时加入`CSRF Token`可增强药效",
      "名场面": "某厂用牛蒡子煮水擦洗防火墙，误拦截了CEO的VPN流量"
    },
    "教主警示": "『牛蒡子如`rm -f`，用前先`--dry-run`』"
  }
}


/*
 * ===========================================================================JAVA
 * 数据协议声明：
 * 1. 本JSON结构属技术演示，所有中医术语均为文化隐喻
 * 2. 实际应用需遵守MIT协议附加条款
 * 3. 企业合作请通过GitHub邮箱联系
 * 
 * 冷溪虎山数字本草实验室 © 2025
 * ===========================================================================JAVA
 */
